components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitConsumer.java

```diff
@@ -0,0 +1,293 @@
+                if (deliveryTag != 0 && !consumer.getEndpoint().isAutoAck()) {
+                    log.trace("Rejecting receipt [delivery_tag={}] with requeue={}", deliveryTag, isRequeueHeaderSet);
+                    if (isRequeueHeaderSet) {
+                        channel.basicReject(deliveryTag, true);
+                    } else {
+                        channel.basicReject(deliveryTag, false);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Bind consumer to channel
+     */
+    public void start() throws IOException {
+        if (channel == null) {
+            throw new IOException("The RabbitMQ channel is not open");
+        }
+        tag = channel.basicConsume(consumer.getEndpoint().getQueue(), consumer.getEndpoint().isAutoAck(), this);
+    }
+
+    /**
+     * Unbind consumer from channel
+     */
+    public void stop() throws IOException, TimeoutException {
+        stopping = true;
+        if (channel == null) {
+            return;
+        }
+        if (tag != null) {
+            channel.basicCancel(tag);
+        }
+        try {
+            channel.close();
+        } catch (TimeoutException e) {
+            log.error("Timeout occured");
+            throw e;
+        }
+    }
+
+    /**
+     * Stores the most recently passed-in consumerTag - semantically, there
+     * should be only one.
+     * 
+     * @see Consumer#handleConsumeOk
+     */
+    public void handleConsumeOk(String consumerTag) {
+        this.consumerTag = consumerTag;
+    }
+
+    /**
+     * Retrieve the consumer tag.
+     * 
+     * @return the most recently notified consumer tag.
+     */
+    public String getConsumerTag() {
+        return consumerTag;
+    }
+
+    /**
+     * No-op implementation of {@link Consumer#handleCancelOk}.
+     * 
+     * @param consumerTag
+     *            the defined consumer tag (client- or server-generated)
+     */
+    public void handleCancelOk(String consumerTag) {
+        // no work to do
+        log.debug("Recieved cancelOk signal on the rabbitMQ channel");
+    }
+
+    /**
+     * No-op implementation of {@link Consumer#handleCancel(String)}
+     * 
+     * @param consumerTag
+     *            the defined consumer tag (client- or server-generated)
+     */
+    public void handleCancel(String consumerTag) throws IOException {
+        // no work to do
+        log.debug("Recieved cancel signal on the rabbitMQ channel");
+    }
+
+    /**
+     * No-op implementation of {@link Consumer#handleShutdownSignal}.
+     */
+    public void handleShutdownSignal(String consumerTag, ShutdownSignalException sig) {
+        log.info("Recieved shutdown signal on the rabbitMQ channel");
+
+        // Check if the consumer closed the connection or something else
+        if (!sig.isInitiatedByApplication()) {
+            // Something else closed the connection so reconnect
+            boolean connected = false;
```

Comment:
This gets called if the channel is closed for any reason and allows it to quickly get reconnected without starting an additional thread.